gfx-examples/src/shadow.rs

```diff
@@ -517,17 +517,15 @@ impl framework::Example for Example {
                 ],
             });
 
-            let uniform_size = mem::size_of::<ForwardUniforms>() as u32;
-            let uniform_buf = device.create_buffer(&wgpu::BufferDescriptor {
-                size: uniform_size,
-                usage: wgpu::BufferUsageFlags::UNIFORM | wgpu::BufferUsageFlags::TRANSFER_DST,
-            });
             let mx_total = Self::generate_matrix(sc_desc.width as f32 / sc_desc.height as f32);
-            let data = ForwardUniforms {
+            let forward_uniforms = ForwardUniforms {
                 proj: *mx_total.as_ref(),
                 num_lights: [lights.len() as u32, 0, 0, 0],
             };
-            uniform_buf.set_sub_data(0, framework::cast_slice(&[data]));
+            let uniform_size = mem::size_of::<ForwardUniforms>() as u32;
+            let uniform_buf =
+                device.create_buffer_mapped(1, wgpu::BufferUsageFlags::UNIFORM)
```

Comment:
we need `TRANSFER_DST` here since we copy into it on `resize()` (and same for cube.rs)